pkg/storage/chunk/client/ibmcloud/cos_object_client_test.go

```diff
@@ -81,3 +124,125 @@ func Test_COSConfig(t *testing.T) {
 		require.Equal(t, []string{tt.cosConfig.BucketNames}, cosClient.bucketNames)
 	}
 }
+
+func Test_DeleteObject(t *testing.T) {
+	tests := []struct {
+		key     string
+		wantErr error
+	}{
+		{
+			"key-4",
+			nil,
+		},
+		{
+			"key-5",
+			errMissingObject,
+		},
+	}
+
+	for _, tt := range tests {
+		cosConfig := COSConfig{
+			BucketNames:     bucket,
+			Endpoint:        "test",
+			Region:          "dummy",
+			AccessKeyID:     "dummy",
+			SecretAccessKey: flagext.SecretWithValue("dummy"),
+			BackoffConfig: backoff.Config{
+				MaxRetries: 1,
+			},
+		}
+		cosClient, err := NewCOSObjectClient(cosConfig, hedging.Config{})
+		require.NoError(t, err)
+		cosClient.cos = newMockCosClient()
+		err = cosClient.DeleteObject(context.Background(), tt.key)
+		if tt.wantErr != nil {
+			require.Equal(t, tt.wantErr.Error(), err.Error())
+			continue
+		}
+		require.NoError(t, err)
+	}
+}
+
+func Test_List(t *testing.T) {
+	tests := []struct {
+		prefix      string
+		delimiter   string
+		storage_obj []client.StorageObject
+		wantErr     error
+	}{
+		{
+			"key",
+			"-",
+			[]client.StorageObject{{Key: "key-1", ModifiedAt: timestamp}, {Key: "key-2", ModifiedAt: timestamp}, {Key: "key-3", ModifiedAt: timestamp}},
+			nil,
+		},
+		{
+			"test",
+			"/",
+			nil,
+			nil,
+		},
+	}
+
+	for _, tt := range tests {
+		cosConfig := COSConfig{
+			BucketNames:     bucket,
+			Endpoint:        "test",
+			Region:          "dummy",
+			AccessKeyID:     "dummy",
+			SecretAccessKey: flagext.SecretWithValue("dummy"),
+			BackoffConfig: backoff.Config{
+				MaxRetries: 1,
+			},
+		}
+		cosClient, err := NewCOSObjectClient(cosConfig, hedging.Config{})
+		require.NoError(t, err)
+		cosClient.cos = newMockCosClient()
+
+		storage_obj, _, err := cosClient.List(context.Background(), tt.prefix, tt.delimiter)
+		if tt.wantErr != nil {
+			require.Equal(t, tt.wantErr.Error(), err.Error())
+			continue
+		}
+		require.NoError(t, err)
+		sort.Slice(storage_obj, func(i, j int) bool {
+			return storage_obj[i].Key < storage_obj[j].Key
+		})
+		require.Equal(t, tt.storage_obj, storage_obj)
+	}
+}
+
+func (cosClient *mockCosClient) DeleteObjectWithContext(ctx context.Context, input *s3.DeleteObjectInput, opts ...request.Option) (*s3.DeleteObjectOutput, error) {
```

Comment:
could you please put all the methods belong to mockCosClient together after the struct is defined?